proto/cosmos/staking/v1beta1/staking.proto

```diff
@@ -347,3 +347,15 @@ message Pool {
     (gogoproto.moretags)   = "yaml:\"bonded_tokens\""
   ];
 }
+
+// InfractionType indicates the infraction type a validator commited.
+enum InfractionType {
+  option (gogoproto.goproto_enum_prefix) = false;
+
+  // UNSPECIFIED defines an empty infraction type.
+  INFRACTION_TYPE_UNSPECIFIED = 0 [(gogoproto.enumvalue_customname) = "InfractionEmpty"];
```

Comment:
Shouldn't the custom name be `InfractionUnspecified`? Also is passing in an unspecified infraction valid? Or should slashing keeper error in this case?

Reply:
I agree that that the `InfractionUnspecified` custom name is more consistent.

>  Also is passing in an unspecified infraction valid? Or should slashing keeper error in this case?

In the SDK, the slashing keeper isn't required to error since the `InfractionType` argument is disabled on the Slash staking keeper's function.
https://github.com/sainoe/cosmos-sdk/blob/7dceda27b5a8f183b109978f50c0f433eccbafa8/x/staking/keeper/slash.go#L24

However, that should be the case in the consumer keeper's implementation of the slash function.